asf_core_data/analysis/data_loading.ipynb

```diff
@@ -0,0 +1,1979 @@
+    "                                          n_samples=5000\n",
+    "                                         )\n",
+    "\n",
+    "prep_epc_dedupl.shape"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 36,
+   "id": "06560798",
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "(19049475, 5)"
+      ]
+     },
+     "execution_count": 36,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "prep_epc_dedupl = load_preprocessed_epc_data(data_path=LOCAL_DATA_DIR, version='preprocessed_dedupl',\n",
+    "                                          usecols=['CURRENT_ENERGY_EFFICIENCY', 'PROPERTY_TYPE', 'BUILT_FORM', 'CONSTRUCTION_AGE_BAND', 'COUNTRY'],\n",
+    "                                              \n",
+    "                                         )\n",
+    "\n",
+    "prep_epc_dedupl.shape"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "b4285a87",
+   "metadata": {},
+   "source": [
+    "### Loading data from directly from S3"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 37,
+   "id": "e725a991",
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "(50000, 5)"
+      ]
+     },
+     "execution_count": 37,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "prep_epc = epc_data.load_preprocessed_epc_data(data_path=\"S3\", version='preprocessed_dedupl',\n",
+    "                                          usecols=['UPRN', 'CURRENT_ENERGY_RATING', 'INSPECTION_DATE',\n",
+    "                                                   'PROPERTY_TYPE', 'CONSTRUCTION_AGE_BAND'],\n",
+    "                                          batch='newest',\n",
+    "                                          n_samples=50000)\n",
+    "\n",
+    "prep_epc.shape"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "ff87ad34",
+   "metadata": {},
+   "source": [
+    "Pro tip: you can filter the data by entry year."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 38,
+   "id": "7c3ac0bc",
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "[2014, 2016, 2017, 2018]"
+      ]
+     },
+     "execution_count": 38,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "prep_epc_2015 = epc_data.filter_by_year(prep_epc, 2018, up_to=True)\n",
+    "sorted(prep_epc_2015[\"INSPECTION_DATE\"].dt.year.unique())"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "7dc630fe",
+   "metadata": {},
+   "source": [
+    "### Preprocess the data\n",
+    "\n",
+    "As preprocessed versions are available on S3 for each batch, this functionionality is optional. Preprocessing can be repeated when the preprocessing pipeline has been updated or if further fields are added. "
```

Comment:
I would probably add just one line explaining what `load_and_preprocess_epc_data()` and `manage_hp_install_dates()` do, to make it easier for the user.

Reply:
Done